src/components/AgentRow.tsx

```diff
@@ -14,32 +16,37 @@ const AgentRow: React.FC<AgentRowProps> = (
 ): JSX.Element => {
   const { agent } = props;
 
-  const { handleAgentCheckboxChange, hiddenAgents } =
+  const { toggleAgentVisibility, hiddenAgents, highlightedAgents } =
     useContext(VisibilityContext);
 
-  const getCheckboxStatus = () => {
+  const getHighlightCheckboxStatus = (): CheckboxState => {
+    if (highlightedAgents[agent.name]?.length === 0) {
+      return CheckboxState.Checked;
+    }
+    return CheckboxState.Unchecked;
+  };
+
+  const getHideCheckboxStatus = (): CheckboxState => {
```

Comment:
I think this now looks a lot better. I would consider sending the two lists into the two checkboxes and having them each handle their own state instead of doing it here. Unless that'll cause a problem I haven't thought of. Since they're all in context it just seems nice to have it as close to the prop doing the stuff as possible

Reply:
I'm fine with that, I'll push it all down, the checkboxes will just need the agent as a prop.

My thinking before was to have checkboxes be just display, and then have `AgentRow` and `DisplayStateRow` but it will be fine either way I think

Reply:
I think the bigger change here will be that making that checkboxes handle more things causes the All Agents checkbox to diverge enough that I might just make a third component for it